### PR TITLE
ci: resolve resulting code base by merging

### PR DESCRIPTION
**Pull request by bergwolf**
If we do rebase, it is required that every commit cannot conflict but a
merge only requires that the final code do not conflict.

Fixes: #2673

### CODE REVIEW

File: .ci/resolve-kata-dependencies.sh [hunk 1]
```diff
@@ -56,8 +56,8 @@ apply_depends_on() {
 		dependency_branch="p${pr_id}"
 		git fetch origin "pull/${pr_id}/head:${dependency_branch}" && \
 			git checkout "${dependency_branch}" && \
-			git rebase "origin/${branch}"
-			# And show what we rebased on top of to aid debugging
+			git merge "origin/${branch}"
```
Comment (jodh-intel) [inline .ci/resolve-kata-dependencies.sh:59]: Could we make this `--ff-only`? Would that work for https://github.com/kata-containers/kata-containers/pull/330 ?

Comment (bergwolf) [reply .ci/resolve-kata-dependencies.sh:59]: We don't need to require --ff-only either. Just simulate a github merge action is enough. And yes, I think this should work for kata-containers/kata-containers#330. waiting to verify.

**Comment by bergwolf** [comment]
It makes sense to use `merge` instead of `rebase` so there is no dependency on kata-containers/kata-containers#330. This is ready to be reviewed/merged.

**Pull request merged**